azurerm/internal/services/policy/tests/policy_set_definition_resource_test.go

```diff
@@ -806,6 +825,54 @@ VALUES
 `, template, data.RandomInteger, data.RandomInteger)
 }
 
+func testAzureRMPolicySetDefinition_customWithDefinitionGroups(data acceptance.TestData) string {
+	template := testAzureRMPolicySetDefinition_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_policy_set_definition" "test" {
+  name         = "acctestPolSet-%d"
+  policy_type  = "Custom"
+  display_name = "acctestPolSet-display-%d"
+
+  parameters = <<PARAMETERS
+    {
+        "allowedLocations": {
+            "type": "Array",
+            "metadata": {
+                "description": "The list of allowed locations for resources.",
+                "displayName": "Allowed locations",
+                "strongType": "location"
+            }
+        }
+    }
+PARAMETERS
+
+  policy_definition_reference {
+    policy_definition_id = azurerm_policy_definition.test.id
+    parameter_values     = <<VALUES
+	{
+      "allowedLocations": {"value": "[parameters('allowedLocations')]"}
+    }
+VALUES
+    group_names          = ["group-1", "group-2"]
+  }
+
+  policy_definition_group {
+    name = "redundant"
+  }
```

Comment:
could we add another test that sets all the properties on the resource? and then go from this config -> all -> back?